jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1614,6 +1614,112 @@ ecma_op_typedarray_define_index_prop (ecma_object_t *obj_p, /**< a TypedArray ob
   return ECMA_VALUE_TRUE;
 } /* ecma_op_typedarray_define_index_prop */
 
+/* Specify the creation of a new TypedArray
+ * object using a constructor function.
+ *
+ * See also: ES11 22.2.4.6
+ *
+ * Used by:
+ *        - ecma_typedarray_species_create
+ *
+ * @return ecma_value_t
+ *
+ */
+ecma_value_t ecma_typedarray_create (ecma_object_t *constructor_p, /**< constructor function */
+                                     ecma_value_t *arguments_list_p, /**< argument list */
+                                     uint32_t arguments_list_len) /**< length of argument list */
+{
+  ecma_number_t num;
+  if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (arguments_list_p[0], &num)))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+  uint32_t length = ecma_number_to_uint32 (num);
```

Comment:
This part is completely wrong.

`ecma_op_function_construct` should be called with `constructor_p, constructor_p, arguments_list_p, arguments_list_len`